tests/cpp/common/test_span.cc

```diff
@@ -399,6 +399,50 @@ TEST(Span, FirstLast) {
       ASSERT_EQ(s[12 + i], last[i]);
     }
 
+    delete [] arr;
+  }
+}
+
+TEST(SpanDeathTest, FirstLast) {
```

Comment:
Is the extraction of these test out of good modularization or out of necessity?

Reply:
This is out of necessity. You cannot run death tests with a RMM pool because you'd then duplicate the pool twice and crash the program with OOM error. I've separated out all death tests so that we can filter them out easily.